### ISSUE TITLE
As a user I want to be able to view/download pdf's and Excel documents

### ISSUE THREAD

**Issue by nilsml** [equinor/energyvision#252]
## Acceptance criteria
1. Documents to be available for user to download
1. Start by putting the documents in Sanity (later to review if documents should be in Brandmaster instead)

## In order to ensure good quality we should (Keep the ones we need for this US)
- [ ] Perform integration testing
- [ ] Perform functional testing
- [ ] Extra focus on usability
- [ ] Test for portability
- [ ] Test for accessibility
- [ ] Create automated acceptance test/smoke test
- [ ] Perform performance and load test
- [ ] Perform user testing
- [ ] Perform security testing

**Comment by nilsml** [comment]
@wenche to check with Sanity if they support zip-files and jpegs as well.

**Comment by wenche** [comment]
Added some icons to make it easier for publishers
![Skjermbilde 2021-04-27 kl. 15.34.31.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/e76d9d10-80d3-44c7-8fa7-fc1f7bb63928)

**Comment by wenche** [comment]
While Sanity does support the upload on a zip archive, we feel that we should investigate security concerns here.
![Skjermbilde 2021-04-27 kl. 15.37.14.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/bba96bdf-dac5-4fc8-a36a-1038d51d119f)

**Comment by wenche** [comment]
Test different types of links: https://energyvision.app.radix.equinor.com/news/4d-seismic-agreements

**Comment by SvSven** [comment]
Works fine from what I can see - I made an issue to follow-up for ZIP files: #283

**Comment by BirteThornquist** [comment]
The heading for the content link section should be left aligned
From Figma:
 ![Skjermbilde 2021-04-29 kl. 15.47.21.png](https://images.zenhubusercontent.com/5fb7acdfad906a977e792e56/6aecc971-0861-4bc0-8a67-39d8745c4255)

From Test
![Skjermbilde 2021-04-29 kl. 15.48.56.png](https://images.zenhubusercontent.com/5fb7acdfad906a977e792e56/9358b5e1-756d-4363-bbc0-e4207179eebe)

Also there seem to be too much spacefrom the Fact Box...

**Comment by wenche** [comment]
Yes, I see it's because we don't get the CSS margin collapse in the case of the last element in the content array. It might actually be a good idea to use a last-child selector in the content array and not add bottom margin to that element. What do you think @vnys 

![Skjermbilde 2021-04-29 kl. 16.14.56.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/48477481-14a7-4e9d-ac8a-66a693003620)

**Comment by wenche** [comment]
@BirteThornquist  Fixed the header alignment, but since the build pipeline is broken it will remain a local secret :) ![Skjermbilde 2021-04-29 kl. 16.27.04.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/9279a93f-6ef6-464f-84ff-0594d8069c55)

**Comment by SvSven** [comment]
Did we agree this morning that this issue is done?

**Comment by petterwildhagen** [comment]
I can upload PDF files. Txt and jpg are blocked

**Comment by wenche** [comment]
@petterwildhagen It's a good thing that you can't upload txt right? For images you'll need to click Upload image :)

**Comment by petterwildhagen** [comment]
Looks good. Tested uploading pdf and xls, xlsx files and this works ok. No accessiblity issues except warning that attachments are an unordered list and  aria=hidden and that PDF and spreadsheets themselves are not acccessible.

**Comment by wenche** [comment]
aria-hidden is for the icons as they are not important for screen readers. :)

**Comment by wenche** [comment]
I added some more validation to avoid issues with empty fields and to reduce the change for making mistakes in Sanity. As we have talked about earlier: We should address validation on a more top-level in the schemas.

![Skjermbilde 2021-05-03 kl. 20.34.38.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/08fddb6d-9c32-489e-9874-d96a3f2d064d)
![Skjermbilde 2021-05-03 kl. 20.34.18.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/ed33ac15-4b53-41a7-ba55-9a259ea999a2)

**Comment by petterwildhagen** [comment]
Should there be a max length for the Name field for the file upload?
![image.png](https://images.zenhubusercontent.com/60376a60b424c72d28d1572f/6d8d3f27-74d2-4ed3-8609-779753f209b7)

**Comment by petterwildhagen** [comment]
In a news article I am trying to upload a file called file_example_XLS_10_3333.xls.
In a previous upload I have uploaded a file called file_example_XLS_10.xls.
On uploading the first file it reverts to uploading the latter
![image.png](https://images.zenhubusercontent.com/60376a60b424c72d28d1572f/a4754fca-0628-43b3-a4dc-b8dd4580c196)

**Comment by wenche** [comment]
@petterwildhagen  Max length for the label could be something to look into, at least on a warning level as with the titles. But I think this should be a separate issue and should be solved by taking an overall look at the whole Sanity input schema and talk about validation in general (min/max/required ++)

**Comment by wenche** [comment]
@petterwildhagen  For the upload bug, if this happens every time and is a consistent bug you should talk to Sanity in the #s-equinor-poc channel as we are just using their ready-made components for this logic. 

I often use a screen recording to show bugs like this. :)

**Comment by SvSven** [comment]
Generally speaking the feedback on using max length as an error validation has been negative, with the argument being that we can leave it up to best judgement. I can see the problem though, and feel like we may actually end up with pretty long file names here if we don't watch out. What do you think @nicholas-haggerty ?

**Comment by petterwildhagen** [comment]
@wenche  I made a post in the #s-equinor-poc channel. Not sure if this is a use bug or Sanity bug. It seems Sanity reverts to the old file name if the contents of the 'new' file is the same as the content of a previously uploaded file that just has a different name

**Comment by wenche** [comment]
@petterwildhagen  If so, that is pretty clever!

**Comment by petterwildhagen** [comment]
@wenche  That is what they are saying on Slack. In trying to upload a duplicate file with a new name it reverts to using the one already uploaded.

**Comment by wenche** [comment]
@petterwildhagen Magic! But it makes sense I guess. Good catch :)

**Comment by petterwildhagen** [comment]
@wenche  Yep, it makes sense, it 'forces' the editors to version files, it is just not possible to upload the same file (same content) using different file names @nicholas-haggerty

**Comment by petterwildhagen** [comment]
Tested with PDF file of 100 Mb which works OK. Tested with uploading XLS, XLSX and PDF files with different names but duplicate names. Sanity reverts to original filename (and so enforces versioning)
This issue can be closed if we have tasks potentially support ZIP files and we leave name length per recommendation (leave it to judgement)

**Comment by nicholas-haggerty** [comment]
> Generally speaking the feedback on using max length as an error validation has been negative, with the argument being that we can leave it up to best judgement. I can see the problem though, and feel like we may actually end up with pretty long file names here if we don't watch out. What do you think @nicholas-haggerty ?

We do have a limitation on the character amount for download button in AEM (see attached)[
![long title download button](https://user-images.githubusercontent.com/58514387/117012069-d3c48e00-acee-11eb-9f6d-41629ea3e390.JPG)
](url)

**Comment by petterwildhagen** [comment]
Created issue #290 to look into limiting the length of the name.
This issue can be closed.
